### PR TITLE
Update Docker build/image

### PR DESCRIPTION
**Pull request by justaugustus**
- Update build container to golang:1.12.4-alpine
- Update dex image to alpine:3.9
- Run dex as non-root user

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

Fixes: #1425

### CODE REVIEW

**Comment by justaugustus** [comment]
cc: @koushik82

File: Makefile [hunk 1]
```diff
@@ -55,7 +55,7 @@ lint: bin/golint
 
 .PHONY: docker-image
 docker-image:
-	@sudo docker build -t $(DOCKER_IMAGE) .
```
Comment (ericchiang) [inline Makefile]: can you revert this change? docker requires (or should require) sudo on linux

Comment (justaugustus) [reply Makefile]: @ericchiang -- Good point. Reverted!
